features/securebackup/impl/src/main/kotlin/io/element/android/features/securebackup/impl/setup/SecureBackupSetupPresenter.kt

```diff
@@ -129,11 +129,11 @@ class SecureBackupSetupPresenter @AssistedInject constructor(
         encryptionService.enableRecoveryProgressStateFlow.collect { enableRecoveryProgress ->
             Timber.tag(loggerTagSetup.value).d("New enableRecoveryProgress: ${enableRecoveryProgress.javaClass.simpleName}")
             when (enableRecoveryProgress) {
-                EnableRecoveryProgress.Unknown,
+                is EnableRecoveryProgress.Starting,
+                is EnableRecoveryProgress.CreatingBackup,
+                is EnableRecoveryProgress.CreatingRecoveryKey,
                 is EnableRecoveryProgress.BackingUp,
-                EnableRecoveryProgress.CreatingBackup,
-                EnableRecoveryProgress.CreatingRecoveryKey ->
-                    Unit
+                is EnableRecoveryProgress.RoomKeyUploadError -> Unit
```

Comment:
I guess this new case `RoomKeyUploadError` should be handled differently, but I can do it in a separate PR (i.e. not block this one).